src/screens/lock/events/LockScreen.events.ts

```diff
@@ -0,0 +1,35 @@
+import { EventEmitter } from 'events'
+import { FlagshipUI } from 'cozy-intent'
+
+import {
+  NormalisedFlagshipUI,
+  ThemeInput,
+  formatTheme,
+  updateStatusBarAndBottomBar
+} from '/libs/intents/setFlagshipUI'
+
+const handleSideEffects = ({
+  bottomTheme,
+  ...parsedIntent
+}: NormalisedFlagshipUI): void => {
+  lockScreenUi.emit('lockscreen:change', parsedIntent)
+  updateStatusBarAndBottomBar(bottomTheme)
+}
+
+export const lockScreenUi = new EventEmitter()
+
+export const setLockScreenUI = (intent: FlagshipUI): Promise<null> => {
+  handleSideEffects(
+    Object.fromEntries(
+      Object.entries({
+        ...intent,
+        bottomTheme: formatTheme(intent.bottomTheme as ThemeInput),
+        topTheme: formatTheme(intent.topTheme as ThemeInput)
+      })
+        .filter(([, v]) => v)
+        .map(([k, v]) => [k, v?.trim()])
```

Comment:
Do we really need this complexity as we have control on the caller code?

Reply:
In my opinion, we do not necessarily have control over the caller code, that's why I added this initially. Many apps aren't typed, and who knows what is coming through during runtime? I don't feel safe if I don't sanitize the input

Reply:
Yes, that is true, I copy pasted from flagshipUI. I guess we can remove it since the scope is fairly limited for now